st2common/st2common/operators.py

```diff
@@ -52,14 +52,20 @@ class UnrecognizedConditionError(Exception):
 
 def search(value, criteria_pattern, criteria_condition, check_function):
     """
-    Search a list of values that match all child criteria. If condition is 'any', return a
-    successful match if any items match all child criteria. If condition is 'all', return a
-    successful match if ALL items match all child criteria.
+    Search a list of values that match all child criteria.
+    Allow comparison of payload items to multiple criteria using different logicial conditions.
+    If condition is 'any', return a successful match if any items match all child criteria.
+    If condition is 'all', return a successful match if ALL items match all child criteria.
+    Other allowed conditions: all2all, all2any, any2any and any2all.
 
     value: the payload list to search
     condition: one of:
-      * any - return true if any items of the list match and false if none of them match
-      * all - return true if all items of the list match and false if any of them do not match
+      * all2all - true if all payload items match all criteria items
+      * all2any - true if all payload items match any criteria items
+      * any2any - true if any payload items match any criteria items
+      * any2all - true if any payload items match all criteria items
+      * any - same as any2all (useful to maintain backward compatibility)
+      * all - same as all2all (useful to maintain backward compatibility)
```

Comment:
There is no need to duplicate `any` and `all` with `any2all` and `all2all`. This would just puzzle the users.
Please add 2 new criteria you'd like as an addition, leaving previous logic and documentation as before.